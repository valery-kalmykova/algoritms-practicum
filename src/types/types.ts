import { ElementStates } from "./element-states";

interface IcirclesData {
  el: string, 
  color: ElementStates,
};

interface IcolumnsData {
  el: number, 
  color: ElementStates,
};

export type {
  IcirclesData,
  IcolumnsData, 
}