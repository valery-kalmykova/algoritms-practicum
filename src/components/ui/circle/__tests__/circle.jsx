import renderer from 'react-test-renderer';
import { Circle } from '../circle';
import { ElementStates } from "../../../../types/element-states";

const letter = 'A';
describe('Circle component tests', () => {

  it('circle without letter', () => {
    const circle = renderer.create(<Circle />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with letter', () => {
    const circle = renderer.create(<Circle text={letter}/>).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with empty head', () => {
    const circle = renderer.create(<Circle head />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with not empty head', () => {
    const circle = renderer.create(<Circle head={<Circle text={letter} />} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with empty tail', () => {
    const circle = renderer.create(<Circle tail />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with not empty tail', () => {
    const circle = renderer.create(<Circle tail={<Circle text={letter} />} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with index', () => {
    const circle = renderer.create(<Circle index={0} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with prop isSmall', () => {
    const circle = renderer.create(<Circle isSmall={true} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with state Default', () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with state Changing', () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

  it('circle with state Modified', () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON()
    expect(circle).toMatchSnapshot()
  })

})