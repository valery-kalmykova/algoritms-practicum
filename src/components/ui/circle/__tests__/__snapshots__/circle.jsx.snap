// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component tests circle with empty head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with empty tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component tests circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with not empty head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with not empty tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component tests circle with prop isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with state Changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with state Default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle with state Modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component tests circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
